x-pack/solutions/observability/plugins/observability/server/services/referenced_panel_manager.ts

```diff
@@ -45,6 +59,50 @@ export class ReferencedPanelManager {
     return panelId ? this.panelsById.get(panelId) : undefined;
   }
 
+  getPanelIndices(panel: DashboardPanel): Set<string> {
+    const res = new Set<string>();
+    let references = isLensVizAttributesForPanel(panel.panelConfig.attributes)
+      ? panel.panelConfig.attributes.references
+      : undefined;
+    if (!references && panel.panelIndex) {
+      references = this.getByIndex(panel.panelIndex)?.references;
+    }
+    references
+      ?.filter((r) => r.name.match(`indexpattern`))
```

Comment:
This is nit, but we can avoid re-declaring this regex for each iteration. Declare it once outside the loop.

```typescript
const indexPatternRegex = /indexpattern/;
references?.filter(r => indexPatternRegex.test(r.name))
```

Reply:
Great catch!

Reply:
Done!